gorm/transaction.go

```diff
@@ -18,14 +19,24 @@ import (
 // This prevents collisions with keys defined in other packages.
 type ctxKey int
 
+// readOnlyDBKey is an unexported type and used to define a key for storing read only db instance in the context.
```

Comment:
Rather than storing the read-only database in the context, cannot we store it in the `Transaction` structure like the existing database? A `nil` value for this variable indicates no read-only database.